src/ipsecondarydiscovery.cc

```diff
@@ -0,0 +1,94 @@
+#include "ipsecondarydiscovery.h"
+
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <chrono>
+
+#include "logging.h"
+
+std::vector<Uptane::SecondaryConfig> IpSecondaryDiscovery::discover() {
+  sendRequest();
+  return waitDevices();
+}
+
+std::vector<Uptane::SecondaryConfig> IpSecondaryDiscovery::waitDevices() {
+  std::vector<Uptane::SecondaryConfig> secondaries;
+  int socket_fd = socket(AF_INET, SOCK_DGRAM, 0);
+  if (socket_fd == -1) {
+    perror("socket creation");
+  }
+
+  int reuse = 1;  // Neded for tests, because client and server runs on the same machine and bind the same port
+  setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &reuse, sizeof reuse);
+
+  struct sockaddr_in recv_addr {};
+  recv_addr.sin_family = AF_INET;
+  recv_addr.sin_port = htons(config_.ipdiscovery_port);
+  recv_addr.sin_addr.s_addr = htonl(INADDR_ANY);
+  if (bind(socket_fd, (struct sockaddr *)&recv_addr, sizeof recv_addr) < 0) {
+    perror("bind");
+  }
+
+  auto start_time = std::chrono::system_clock::now();
+  struct timeval tv;
+  tv.tv_sec = config_.ipdiscovery_wait_seconds;
+  tv.tv_usec = 0;
+  setsockopt(socket_fd, SOL_SOCKET, SO_RCVTIMEO, (const char *)&tv, sizeof(tv));
+
+  char rbuf[2000] = {};
+  int recieved = 0;
+  while ((recieved = recv(socket_fd, rbuf, sizeof(rbuf) - 1, 0)) != -1) {
+    std::string data(std::move(rbuf), recieved);
+    try {
+      int type = 0;
+      Uptane::SecondaryConfig conf;
```

Comment:
The secondary config should probably contain the device ip so that we can communicate with it on the regular interface later.